### PR TITLE
docs: Erc20Token

### PR DESCRIPTION
**Pull request by sogunshola**
technical docs for ERC20 token

---
name: 'Pull request'
about: A new pull request
---

### Checklist

- [X] I am not disclosing a [vulnerability](https://github.com/MoralisWeb3/Moralis-JS-SDK/blob/main/SECURITY.md).
- [X] My code is conform the [code style](https://github.com/MoralisWeb3/Moralis-JS-SDK/blob/main/CODE_STYLE.md)
- [X] I have made corresponding changes to the documentation
- [X] I have updated Typescript definitions when needed

### Issue Description

Related issue: #`FILL_THIS_OUT`

### CODE REVIEW

File: packages/evmUtils/src/dataTypes/Erc20/Erc20.ts [hunk 1]
```diff
@@ -3,13 +3,35 @@ import { EvmAddress } from '../EvmAddress';
...
  * The Erc20Token class is a MoralisData that references to a Erc20 Token
  * It holds data about the data and metadata of an Erc20 token
+ *
+ * @category DataType
  */
 export class Erc20Token implements MoralisDataObject {
+  /**
+   *  Create a new instance of Erc20Token from any valid Erc20Token input
+   *
+   * @param value - the Erc20Tokenish type
```
Comment (ErnoW) [inline packages/evmUtils/src/dataTypes/Erc20/Erc20.ts:21]: Please link to Erc20Tokenish, with {@link } (I think that should work)

File: packages/evmUtils/src/dataTypes/Erc20/Erc20.ts [hunk 2]
```diff
@@ -27,14 +49,17 @@ export class Erc20Token implements MoralisDataObject {
...
-  static create(value: Erc20Tokenish) {
-    if (value instanceof Erc20Token) {
-      return value;
-    }
-
-    return new Erc20Token(value);
-  }
-
+  /**
+   * Compares two Erc20Token instances
```
Comment (ErnoW) [inline packages/evmUtils/src/dataTypes/Erc20/Erc20.ts:53]: Add information that we do equality check based on contractAddress and the chain

File: packages/evmUtils/src/dataTypes/Erc20/Erc20.ts [hunk 3]
```diff
@@ -50,56 +75,148 @@ export class Erc20Token implements MoralisDataObject {
...
+   * @example
+   * ```ts
+   * token.equals(value);
+   * ```
+   */
+  equals(value: Erc20Tokenish): boolean {
+    return Erc20Token.equals(this, value);
+  }
+
+  toJSON() {
```
Comment (ErnoW) [inline packages/evmUtils/src/dataTypes/Erc20/Erc20.ts:92]: Missing docs

File: packages/evmUtils/src/dataTypes/Erc20/Erc20.ts [hunk 4]
```diff
@@ -50,56 +75,148 @@ export class Erc20Token implements MoralisDataObject {
...
+   * ```ts
+   * token.decimals;
+   * ```
+   */
   get decimals() {
     return this._value.decimals;
   }
 
+  /**
+   * Returns the name of the token.
```
Comment (ErnoW) [inline packages/evmUtils/src/dataTypes/Erc20/Erc20.ts:140]: I think you can use `@returns the name of the token here`

Comment (ErnoW) [reply packages/evmUtils/src/dataTypes/Erc20/Erc20.ts:140]: Also in these cases add a comment with example of the output

**Review by ErnoW — CHANGES_REQUESTED** [review]

File: packages/evmUtils/src/dataTypes/Erc20/Erc20.ts (hunk 1 above)
Comment (sogunshola) [reply packages/evmUtils/src/dataTypes/Erc20/Erc20.ts:21]: The @param decorator already links the params type
Screenshot 2022-08-27 at 07 49 59

**Review by ErnoW — APPROVED** [review]

**Pull request merged**